<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ShapeRenderQtClass</class>
 <widget class="QMainWindow" name="ShapeRenderQtClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1100</width>
    <height>928</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>ModelRepairQt</string>
  </property>
  <property name="windowIcon">
   <iconset resource="shaperenderqt.qrc">
    <normaloff>:/ShapeRenderQt/ref/square.png</normaloff>:/ShapeRenderQt/ref/square.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1100</width>
     <height>23</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>10</pointsize>
    </font>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="action_New"/>
    <addaction name="action_Open"/>
    <addaction name="action_Save"/>
    <addaction name="separator"/>
    <addaction name="action_Exit"/>
   </widget>
   <widget class="QMenu" name="menu_View">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <addaction name="action_Points"/>
    <addaction name="action_Edges"/>
    <addaction name="action_Hidden_Lines"/>
    <addaction name="action_Flat"/>
    <addaction name="action_Smooth"/>
    <addaction name="action_Textures"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_View"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="windowTitle">
    <string>File</string>
   </property>
   <property name="iconSize">
    <size>
     <width>36</width>
     <height>36</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_New"/>
   <addaction name="action_Open"/>
   <addaction name="action_Save"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="viewToolBar">
   <property name="windowTitle">
    <string>View</string>
   </property>
   <property name="iconSize">
    <size>
     <width>36</width>
     <height>36</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_Points"/>
   <addaction name="action_Edges"/>
   <addaction name="action_Hidden_Lines"/>
   <addaction name="action_Flat"/>
   <addaction name="action_Smooth"/>
   <addaction name="action_Textures"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget_QEM">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="font">
    <font>
     <pointsize>9</pointsize>
    </font>
   </property>
   <property name="acceptDrops">
    <bool>false</bool>
   </property>
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="windowTitle">
    <string>Mesh Simplification Panel</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QGroupBox" name="groupBox_CrestLine">
       <property name="title">
        <string>Crest Lines and Vertex Quality Confidence</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QCheckBox" name="checkBox_ShowCrestLine">
          <property name="text">
           <string>Show Crest Lines</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_Ridgeness">
          <property name="text">
           <string>Ridgeness</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QScrollBar" name="horizontalScrollBar_Ridgeness">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="singleStep">
           <number>1</number>
          </property>
          <property name="pageStep">
           <number>10</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_Sphericalness">
          <property name="text">
           <string>Sphericalness</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QScrollBar" name="horizontalScrollBar_Sphericalness">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_Cyclideness">
          <property name="text">
           <string>Cyclideness</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QScrollBar" name="horizontalScrollBar_Cyclideness">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkBox_UseCrestLine">
          <property name="text">
           <string>Use Crest Line Weight: eta = </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEdit_eta">
          <property name="text">
           <string>3</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkBox_UseVertexQuality">
          <property name="text">
           <string>Use Vertex Quality Weight: gamma = </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEdit_gamma">
          <property name="text">
           <string>3</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_VertexQualityThreshold">
          <property name="text">
           <string>Vertex Quality Threshold</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEdit_VertexQualityThreshold">
          <property name="text">
           <string>0.05</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_ComputeWeights">
          <property name="text">
           <string>Compute Vertex Weights</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_QEM">
       <property name="title">
        <string>Mesh Simplification</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QLabel" name="label_Vertex_Number">
          <property name="text">
           <string>Final Vertex Number</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEdit_FinalVertexNumber">
          <property name="text">
           <string>1000</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_ReductionRatio">
          <property name="text">
           <string>Reduction Ratio (0-1)</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEdit_ReductionRatio">
          <property name="text">
           <string>0.01</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkBox_ShowClusters">
          <property name="text">
           <string>Show Clusters</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_AllSteps">
          <property name="text">
           <string>QEM-based Simplification</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_SeparateSteps">
          <property name="title">
           <string>Separate Steps</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QPushButton" name="pushButton_QEM">
             <property name="text">
              <string>QSlim</string>
             </property>
             <property name="flat">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_QEM_Optimization">
             <property name="text">
              <string>Swap</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_SplitClusters">
             <property name="text">
              <string>Split Clusters</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_ReQEM">
             <property name="text">
              <string>Re-QEM After Split</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_SplitNonManifold">
             <property name="text">
              <string>Split Non-Manifold Clusters</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_CreateConnectivity">
             <property name="text">
              <string>Create New Connectivity</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_ReverseNormals">
             <property name="text">
              <string>Reverse Normals</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="action_Open">
   <property name="icon">
    <iconset resource="shaperenderqt.qrc">
     <normaloff>:/ShapeRenderQt/ref/open.png</normaloff>:/ShapeRenderQt/ref/open.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action_Exit">
   <property name="icon">
    <iconset resource="shaperenderqt.qrc">
     <normaloff>:/ShapeRenderQt/ref/exit.png</normaloff>:/ShapeRenderQt/ref/exit.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="action_Save">
   <property name="icon">
    <iconset resource="shaperenderqt.qrc">
     <normaloff>:/ShapeRenderQt/ref/save.png</normaloff>:/ShapeRenderQt/ref/save.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_New">
   <property name="icon">
    <iconset resource="shaperenderqt.qrc">
     <normaloff>:/ShapeRenderQt/ref/new.png</normaloff>:/ShapeRenderQt/ref/new.png</iconset>
   </property>
   <property name="text">
    <string>&amp;New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="action_QEM">
   <property name="text">
    <string>QEM</string>
   </property>
  </action>
  <action name="action_QEM_Optimization">
   <property name="text">
    <string>QEM Optimization</string>
   </property>
  </action>
  <action name="action_Points">
   <property name="icon">
    <iconset resource="shaperenderqt.qrc">
     <normaloff>:/ShapeRenderQt/ref/points.png</normaloff>:/ShapeRenderQt/ref/points.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Points</string>
   </property>
  </action>
  <action name="action_Edges">
   <property name="icon">
    <iconset resource="shaperenderqt.qrc">
     <normaloff>:/ShapeRenderQt/ref/wire.png</normaloff>:/ShapeRenderQt/ref/wire.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Edges</string>
   </property>
  </action>
  <action name="action_Hidden_Lines">
   <property name="icon">
    <iconset resource="shaperenderqt.qrc">
     <normaloff>:/ShapeRenderQt/ref/backlines.png</normaloff>:/ShapeRenderQt/ref/backlines.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Hidden Lines</string>
   </property>
  </action>
  <action name="action_Flat">
   <property name="icon">
    <iconset resource="shaperenderqt.qrc">
     <normaloff>:/ShapeRenderQt/ref/flat.png</normaloff>:/ShapeRenderQt/ref/flat.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Flat</string>
   </property>
  </action>
  <action name="action_Smooth">
   <property name="icon">
    <iconset resource="shaperenderqt.qrc">
     <normaloff>:/ShapeRenderQt/ref/smooth.png</normaloff>:/ShapeRenderQt/ref/smooth.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Smooth</string>
   </property>
  </action>
  <action name="action_Textures">
   <property name="icon">
    <iconset resource="shaperenderqt.qrc">
     <normaloff>:/ShapeRenderQt/ref/textures.png</normaloff>:/ShapeRenderQt/ref/textures.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Textures</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="shaperenderqt.qrc"/>
 </resources>
 <connections/>
</ui>
